### ISSUE TITLE
Using --tied-embeddings-all with 2 different vocabs

### ISSUE THREAD

**Issue by okgrammer** [marian-nmt/marian-dev#502]
Hello, I noticed that in Marian you could specify `--tied-embeddings-all` with 2 different vocab files. It will not raise errors as long as the `--dim-vocabs` for both vocabs are the same.

May I check what is the behaviour of Marian in this case? Does it mean it's possible that the embeddings for possibly irrelevant words between source and target with same index are tied? Perhaps it's good to union the vocabs in this case before creating shared embeddings (or give some warnings to the user..)

**Comment by emjotde** [comment]
I have been wondering about that for a while. When the number of items in the vocab matches they will indeed tie across indices. I always wanted to check what the effects of this are. I would not be surprised if it turned out to be barely or not at all harmful, but another interesting way to do model compression.

**Comment by okgrammer** [comment]
When I experimented with these settings on my data set, I observed slower convergence compared to both using `--tied-embeddings` with 2 vocabs and `--tied-embeddings-all`  with 1 vocab, e.g. previously it converged after 30 epochs, now only after 37. The BLEU score is similar to `--tied-embeddings` with 2 vocabs, and they are about 1 point lower than the BLEU score when using `--tied-embeddings-all` with 1 vocab.
